nntrainer/src/parse_util.cpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (C) 2020 Samsung Electronics Co., Ltd. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @file	parse_util.cpp
+ * @date	08 April 2020
+ * @brief	This is collection of math functions
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#include "parse_util.h"
+#include <array>
+#include <assert.h>
+
+namespace nntrainer {
+
+/**
+ * @brief     compare character to remove case sensitivity
+ * @param[in] c1 characer #1
+ * @param[in] c2 characer #2
+ * @retval    boolean true if they are same
+ */
+bool compareChar(char &c1, char &c2) {
```

Comment:
How about `const char` or remove `alias`?  There seems side effect.

Reply:
please be more specific. what kind of side effect do you mean?

Reply:
My apologies. My point was `c1` and `c2` looks like they could be changed from this function. 

I know looking at the code it is perfectly safe. 
However, from the signature, it is not guaranteed that this function does not modify input argument while `bool compareChar(char c1, char c2)` guarantees 

For example  [`boost::toupper`](https://www.boost.org/doc/libs/1_41_0/doc/html/boost/algorithm/to_upper.html) modifies input.

Not important though :(